Rocket.Chat/Models/Message/UnmanagedMessage.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  UnmanagedMessage.swift
+//  Rocket.Chat
+//
+//  Created by Filipe Alvarenga on 19/09/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+import DifferenceKit
+
+struct UnmanagedMessage: UnmanagedObject, Equatable {
+    typealias Object = Message
+    var identifier: String
+    var managedObject: Message
+    var text: String
+    var attachments: [Attachment]
+    var user: UnmanagedUser?
+}
+
+extension UnmanagedMessage {
+    init(_ message: Message) {
+        managedObject = message
+        identifier = message.identifier ?? "" // FA NOTE: We must check if we have a valid identifier before calling this init. If a message doesn't have an identifier we should consider it invalid.
```

Comment:
We can maybe change the protocol to use an optional init..?

Reply:
@Sameesunkaria that may be a good solution. We just have to know if we need the `identifier` to be optional in the first place. If we make `identifier` a required (non-optional) property we won't need to make this init optional. 

I mean, if we have a valid `managed` message we should be able to create an `unmanaged` from it.

Reply:
@filipealva There are other use cases for it as well. We wouldn't have to check for if an object is invalidated, if we can just return nil on the unmanaged object.

Reply:
@Sameesunkaria it makes sense 👍

@rafaelks can you handle this change when working on the data controller?

Reply:
Yes, implemented in my PR (already merged) 👍